cppcache/test/CacheTest.cpp

```diff
@@ -61,3 +62,10 @@ TEST(CacheTest, close) {
   EXPECT_THROW(cache.readyForEvents(), CacheClosedException);
   EXPECT_THROW(cache.rootRegions(), CacheClosedException);
 }
+
+TEST(CacheTest, changeLogLevel) {
+  auto cache = CacheFactory{}.set("log-level", "info").create();
+  ASSERT_EQ(cache.getLogLevel(), LogLevel::Info);
```

Comment:
What happens if you call `cache.getSystemProperties.getLogLevel()` after changing it via `Cache`?

Reply:
Exactly - I was thinking about this earlier this morning.  We can still get into the kind of weird situation where the `geode.properties` file says `log-level` is one value but in fact it's another, and the `SystemProperties` object is also in disagreement about `log-level`.  I don't think the first situation is worth worrying about - unless I'm very mistaken, you'll be in this state even when you set `log-level` on the cache factory to a custom value.  The second, however, it seems to me we should be able to prevent.

Reply:
I have changed the `setLogLevel` method to also update the log level on the `SystemProperties` object.